compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ReachabilityAnalyzer.java

```diff
@@ -744,25 +744,25 @@ public void visit(BLangLetExpression letExpression, AnalyzerData data) {
         data.statementReturnsPanicsOrFails = returnStateBefore;
     }
 
-    private void checkStatementExecutionValidity(BLangStatement statement, AnalyzerData data) {
+    private void checkStatementExecutionValidity(BLangNode node, AnalyzerData data) {
         if (data.skipFurtherAnalysisInUnreachableBlock) {
             return;
         }
-        checkUnreachableCode(statement.pos, data);
-        checkConditionInWhileOrIf(statement, data);
+        checkUnreachableCode(node.pos, data);
+        checkConditionInWhileOrIf(node, data);
     }
 
-    private void checkConditionInWhileOrIf(BLangStatement statement, AnalyzerData data) {
-        switch (statement.getKind()) {
+    private void checkConditionInWhileOrIf(BLangNode node, AnalyzerData data) {
+        switch (node.getKind()) {
             case WHILE:
                 data.booleanConstCondition = ConditionResolver.checkConstCondition(types, symTable,
-                        ((BLangWhile) statement).expr);
+                        ((BLangWhile) node).expr);
                 break;
             case IF:
-                data.unreachableBlock = statement.parent != null && statement.parent.getKind() == NodeKind.IF
+                data.unreachableBlock = node.parent != null && node.parent.getKind() == NodeKind.IF
                         && data.booleanConstCondition == symTable.trueType;
                 data.booleanConstCondition = ConditionResolver.checkConstCondition(types, symTable,
-                        ((BLangIf) statement).expr);
+                        ((BLangIf) node).expr);
                 break;
```

Comment:
> Since `BLangWorkerAsyncSendExpr` is a `BLangExpression` now, we don't need to check statement validity for `asynSendExpr`. Therefore we can get rid of calling this function here.

 As per the above comment we don't need to change these methods as we only do this for statements.

Reply:
Updated 40cc512e09d1ce6989618c1c7392ab833ab772b5